connectivity/check/test.go

```diff
@@ -294,7 +294,10 @@ func (t *Test) finalize() {
 	t.Debug("Finalizing Test", t.Name())
 
 	for _, f := range t.finalizers {
-		if err := f(); err != nil {
+		// Use a detached context to make sure this call is not affected by
+		// context cancellation. Usually, finalization (e.g., netpol removal)
+		// needs to happen even when the user interrupted the program.
+		if err := f(context.TODO()); err != nil {
```

Comment:
I wonder if it could make sense to add a timeout to prevent finalizers from hanging forever in case of e.g., network connectivity problems.

Reply:
What could be a reasonable timeout? 5min?

Reply:
Hmm, I was actually thinking about a way lower value (something between 1 and 5 seconds), but it may be indeed to low if we are waiting for any operation to actually happen (like in case of policies removal). Maybe 1 minute?